src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/IMergeBlockProducer.cs

```diff
@@ -0,0 +1,12 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Consensus;
+
+namespace Nethermind.Merge.Plugin.BlockProduction;
+
+public interface IMergeBlockProducer : IBlockProducer
```

Comment:
I don't quite get why we need this interface. Would it be possible to remove it and pass the required objects to its single call site (`MergePlugin.BlockProducer`)?

Reply:
It doesn't, I did the interface thing because I thought other Merge-like BlockProducers like the AuraMergeBlockProducer would need this logic and they might not inherit directly from the concrete MergeBlockProducer. However it turns out they did inherit from MergeBlockProducer in the end. I left the interface just in case it's needed in a future but could also be removed tbh

Reply:
Up to you. I would personally remove it given the additional indirection. We can reintroduce it if needed.